[inputs.gnmi] panic when empty update with deletes

### Relevant telegraf.conf

```toml
[[inputs.gnmi]]
  addresses     = ["192.168.10.1"]
  redial        = "10s"

  [[inputs.gnmi.subscription]]
    name = "interface_state_change"
    path = "huawei-ifm:ifm/interfaces/interface/dynamic/oper-status"
    subscription_mode = "on_change"
```

### Logs from Telegraf

```text
2025-07-31T07:17:56Z I! Loading config: /etc/telegraf/telegraf-on-change.conf
2025-07-31T07:17:56Z I! Config watcher started for /etc/telegraf/telegraf-on-change.conf
{"time":"2025-07-31T07:17:56.759021369Z","level":"INFO","msg":"Loading config: /etc/telegraf/telegraf-on-change.conf"}
{"time":"2025-07-31T07:17:56.76488263Z","level":"INFO","msg":"Config watcher started for /etc/telegraf/telegraf-on-change.conf"}
{"time":"2025-07-31T07:17:56.764994381Z","level":"INFO","msg":"Starting Telegraf 0.0.1 (customized) brought to you by InfluxData the makers of InfluxDB"}
{"time":"2025-07-31T07:17:56.765002467Z","level":"INFO","msg":"Available plugins: 7 inputs, 1 aggregators, 5 processors, 1 parsers, 3 outputs, 0 secret-stores"}
{"time":"2025-07-31T07:17:56.765007576Z","level":"INFO","msg":"Loaded inputs: gnmi internal"}
{"time":"2025-07-31T07:17:56.76500994Z","level":"INFO","msg":"Loaded aggregators:"}
{"time":"2025-07-31T07:17:56.765012128Z","level":"INFO","msg":"Loaded processors:"}
{"time":"2025-07-31T07:17:56.765014672Z","level":"INFO","msg":"Loaded secretstores:"}
{"time":"2025-07-31T07:17:56.765018553Z","level":"INFO","msg":"Loaded outputs: health kafka (2x)"}
{"time":"2025-07-31T07:17:56.765021494Z","level":"INFO","msg":"Tags enabled: host=huawei-gnmi-collector-ne8000-m14-6798613e-7b4c5c5fb7-9rvc7"}
{"time":"2025-07-31T07:17:56.765033648Z","level":"INFO","msg":"[agent] Config: Interval:5m0s, Quiet:false, Hostname:\"huawei-gnmi-collector\", Flush Interval:10s"}
{"time":"2025-07-31T07:17:56.767599316Z","level":"INFO","msg":"Listening on http://[::]:8082","plugin":"health","category":"outputs"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xdc0116]

goroutine 85 [running]:
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*pathInfo).append(0xc000a12050, {0xc0005c3940, 0x1, 0x6?})
        github.com/influxdata/telegraf/plugins/inputs/gnmi/path.go:134 +0x3d6
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*handler).handleSubscribeResponseUpdate(0xc0005c3d80, {0x1943900, 0xc000278d40}, 0xc0006d2030, {0x0, 0x0, 0x5?})
        github.com/influxdata/telegraf/plugins/inputs/gnmi/handler.go:179 +0x59d
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*handler).subscribeGNMI(0xc0005c3d80, {0x1934558, 0xc00033cd50}, {0x1943900, 0xc000278d40}, 0x1?, 0xc000309810)
        github.com/influxdata/telegraf/plugins/inputs/gnmi/handler.go:123 +0xc45
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).Start.func1({0xc0005224f8, 0x11})
        github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:313 +0x785
created by github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).Start in goroutine 1
        github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:280 +0x359
```

### System info

Telegraf v1.35.3 customized

### Docker

_No response_

### Steps to reproduce

1. Run telegraf
2. Delete interface from Huawei device
3. Observe that Telegraf crashes.
...


### Expected behavior

Ignores empty updates

### Actual behavior

Crashes due to panic

### Additional info

This might also be bug in device, as seems a bit weird that empty update is included. Probably sub-interface deletion should't trigger oper-status deletion event

--- a/plugins/inputs/gnmi/handler.go
+++ b/plugins/inputs/gnmi/handler.go
@@ -176,6 +176,14 @@ func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, respon
 	// add all available tags to the metrics later.
 	var valueFields []updateField
 	for _, update := range response.Update.Update {
+		if update.Path == nil {
+			continue
+		}
+
+		if len(update.Path.Elem) == 0 && prefix.empty() {
+			continue
+		}
+
 		fullPath := prefix.append(update.Path)
 		if h.enforceFirstNamespaceAsOrigin {
 			prefix.enforceFirstNamespaceAsOrigin()

--- a/plugins/inputs/gnmi/testcases/issue_17412/expected.out
+++ b/plugins/inputs/gnmi/testcases/issue_17412/expected.out
@@ -1,0 +1,1 @@
+interface_state_change,name=Eth-Trunk5.1234,source=127.0.0.1 oper_status="down" 1753972839609000000

--- a/plugins/inputs/gnmi/testcases/issue_17412/responses.json
+++ b/plugins/inputs/gnmi/testcases/issue_17412/responses.json
@@ -1,0 +1,71 @@
+[
+  {
+    "update": {
+      "timestamp": "1753972805570000000",
+      "update": [
+        {}
+      ],
+      "delete": [
+        {
+          "elem": [
+            {
+              "name": "huawei-ifm:ifm"
+            },
+            {
+              "name": "interfaces"
+            },
+            {
+              "name": "interface",
+              "key": {
+                "name": "Eth-Trunk5.1234"
+              }
+            },
+            {
+              "name": "dynamic"
+            },
+            {
+              "name": "oper-status"
+            }
+          ]
+        }
+      ]
+    }
+  },
+  {
+    "update": {
+      "timestamp": "1753972839609000000",
+      "update": [
+        {
+          "path": {
+            "elem": [
+              {
+                "name": "huawei-ifm:ifm"
+              },
+              {
+                "name": "interfaces"
+              },
+              {
+                "name": "interface",
+                "key": {
+                  "name": "Eth-Trunk5.1234"
+                }
+              },
+              {
+                "name": "dynamic"
+              },
+              {
+                "name": "oper-status"
+              }
+            ]
+          },
+          "val": {
+            "jsonVal": "ImRvd24i"
+          }
+        }
+      ],
+      "delete": [
+        {}
+      ]
+    }
+  }
+]

--- a/plugins/inputs/gnmi/testcases/issue_17412/telegraf.conf
+++ b/plugins/inputs/gnmi/testcases/issue_17412/telegraf.conf
@@ -1,0 +1,8 @@
+[[inputs.gnmi]]
+  addresses     = ["dummy"]
+  redial        = "10s"
+
+  [[inputs.gnmi.subscription]]
+    name = "interface_state_change"
+    path = "huawei-ifm:ifm/interfaces/interface/dynamic/oper-status"
+    subscription_mode = "on_change"

